src/utils/post_utils.js

```diff
@@ -175,3 +183,98 @@ export function comparePosts(a, b) {
 
     return 0;
 }
+
+export function combineSystemPost(postsIds = [], posts = {}) {
+    if (postsIds.length === 0) {
+        return {postsForChannel: postsIds, nextPosts: posts};
+    }
+
+    const postsForChannel = [];
+    const nextPosts = {...posts};
+
+    let userActivitySystemPosts = [];
+    let systemPostIds = [];
+    let createAt;
+    let combinedPostId;
+
+    postsIds.forEach((p, i) => {
+        const channelPost = posts[p];
+        if (
+            channelPost.delete_at === 0 &&
+            (isUserActivityPost(channelPost.type) || channelPost.type === Posts.POST_TYPES.COMBINED_USER_ACTIVITY)
+        ) {
+            if (!createAt || createAt > channelPost.create_at) {
+                createAt = channelPost.create_at;
+            }
+
+            if (isUserActivityPost(channelPost.type)) {
+                userActivitySystemPosts = [...userActivitySystemPosts, channelPost];
+                systemPostIds.push(channelPost.id);
+
+                if (nextPosts[channelPost.id]) {
+                    nextPosts[channelPost.id] = {...channelPost, state: Posts.POST_DELETED, delete_at: 1};
+                }
+            } else if (channelPost.type === Posts.POST_TYPES.COMBINED_USER_ACTIVITY) {
+                userActivitySystemPosts = [...userActivitySystemPosts, ...channelPost.user_activity_posts];
+                systemPostIds = [...systemPostIds, ...channelPost.system_post_ids];
+
+                combinedPostId = channelPost.id;
+            }
+        }
+
+        if (channelPost.type === '') {
+            if (userActivitySystemPosts.length > 0) {
+                const combinedPost = {
+                    id: combinedPostId || generateId(),
+                    root_id: '',
+                    type: Posts.POST_TYPES.COMBINED_USER_ACTIVITY,
+                    message: '',
```

Comment:
You mentioned that having no message will make this not work with the mobile app temporarily, but what if you make the combined post's message just the messages of each system posts joined together? That would help solve that case and also make it easier if anyone else is building a client with this

Reply:
I think one of my worry is the localization but I'll see what I can do here.

Reply:
I think it's fine if it's a simpler message like "Harrison joined the channel. Harrison left the channel." here, but it'll make sure that we have something to fall back to

Reply:
hmmm, forgot about this... will update after PM review.